api/src/controllers/team.js

```diff
@@ -92,21 +92,23 @@ function getTeam(req, res, next) {
  * @param {*} res response object - updated team object
  * @returns 201: returns updated team details
  */
-async function addTeam(req, res, next) {
+function addTeam(req, res, next) {
   Team.findOne({ email: req.body.email })
     .then((foundTeam) => {
       if (foundTeam) {
         return res.status(400).send('Email had been registered');
       }
+      bcrypt.genSalt().then((salt) => 
+        bcrypt.hash(req.body.password, salt).then((hashedPassword) => {
+          const hashedTeam = { ...req.body, password: hashedPassword };
+          Team.create(hashedTeam)
+            .then(() => {
+              return res.status(201).json('Team has been created successfully')
+            })
+        })
+      );
     })
     .catch((err) => next(fillErrorObject(500, 'Server error', [err.errors])));
-
-  const salt = await bcrypt.genSalt();
-  const hashedPassword = await bcrypt.hash(req.body.password, salt);
-  const hashedTeam = { ...req.body, password: hashedPassword };
-  Team.create(hashedTeam)
-    .then((createdTeam) => res.status(201).json(createdTeam._id))
-    .catch((err) => next(fillErrorObject(500, 'Server error', [err.errors])));
```

Comment:
Add a try-catch around this

Reply:
The winder-break team has created an error handler function, so any error the function cannot be handled will be passed to the error handler. So I think try catch blocks will no longer need. The error handler function is in api/server.js currently. 
A bit of info: https://expressjs.com/en/guide/error-handling.html

Reply:
Makes sense. Resolving.